commercetools-models/src/test/java/io/sphere/sdk/subscriptions/commands/SubscriptionDeleteCommandIntegrationTest.java

```diff
@@ -1,24 +1,54 @@
 package io.sphere.sdk.subscriptions.commands;
 
 import io.sphere.sdk.subscriptions.Subscription;
+import io.sphere.sdk.subscriptions.SubscriptionDraftBuilder;
+import io.sphere.sdk.subscriptions.SubscriptionFixtures;
 import io.sphere.sdk.subscriptions.queries.SubscriptionQuery;
-import io.sphere.sdk.test.IntegrationTest;
 import org.junit.Test;
 
+import java.util.function.Supplier;
+
 import static io.sphere.sdk.subscriptions.SubscriptionFixtures.*;
 import static io.sphere.sdk.test.SphereTestUtils.assertEventually;
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * Integration tests for {@link SubscriptionDeleteCommand}.
  */
-public class SubscriptionDeleteCommandIntegrationTest extends IntegrationTest {
+public class SubscriptionDeleteCommandIntegrationTest extends AbstractQueueIntegrationTest {
+
+
+
+    @Test
+    public void deleteByKeyAzureQueue() throws Exception {
+        assumeHasAzureSBEnv();
+        deleteByKey(SubscriptionFixtures::azureServiceBusSubscriptionDraftBuilder);
+
+    }
+
+    @Test
+    public void deleteByIdAzureQueue() throws Exception {
+        assumeHasAzureSBEnv();
+        deleteByID(SubscriptionFixtures::azureServiceBusSubscriptionDraftBuilder);
+
+    }
 
     @Test
     public void deleteByKeyIronMq() throws Exception {
         assumeHasIronMqEnv();
+        deleteByKey(SubscriptionFixtures::ironMqSubscriptionDraftBuilder);
+
+    }
+
+    @Test
+    public void deleteByIdIronMq() throws Exception {
+        assumeHasIronMqEnv();
+        deleteByID(SubscriptionFixtures::ironMqSubscriptionDraftBuilder);
+
+    }
 
-        final Subscription subscription = createSubscription(client(), withCategoryChanges(ironMqSubscriptionDraftBuilder()));
+    public void deleteByKey(Supplier<SubscriptionDraftBuilder> subscriptionDraftBuilderSupplier) throws Exception {
```

Comment:
Nice 😄 I really like your idea to extract the test methods and then to pass in a subscription draft builder supplier 👍